ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthSendRawTransactionConditional.java

```diff
@@ -0,0 +1,166 @@
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.SendRawTransactionConditionalParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.eth.transactions.TransactionPool;
+import org.hyperledger.besu.evm.internal.StorageEntry;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class EthSendRawTransactionConditional extends AbstractEthSendRawTransaction {
+  protected final Supplier<BlockchainQueries> blockchainQueries;
+
+  public static final long MAX_CONDITIONS = 100; // TODO make this configurable
+
+  public EthSendRawTransactionConditional(
+      final BlockchainQueries blockchainQueries, final TransactionPool transactionPool) {
+    this(Suppliers.ofInstance(blockchainQueries), Suppliers.ofInstance(transactionPool), false);
+  }
+
+  public EthSendRawTransactionConditional(
+      final Supplier<BlockchainQueries> blockchainQueries,
+      final Supplier<TransactionPool> transactionPool,
+      final boolean sendEmptyHashOnInvalidBlock) {
+    super(transactionPool, sendEmptyHashOnInvalidBlock);
+    this.blockchainQueries = blockchainQueries;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ETH_SEND_RAW_TRANSACTION_CONDITIONAL.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    if (requestContext.getRequest().getParamLength() < 1) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(), JsonRpcError.INVALID_PARAMS);
+    }
+    final Optional<JsonRpcErrorResponse> errorResponse = checkConditions(requestContext);
+    if (errorResponse.isEmpty()) {
+      return processRawTransaction(requestContext);
+    } else {
+      return errorResponse.get();
+    }
+  }
+
+  private Optional<JsonRpcErrorResponse> checkConditions(
+      final JsonRpcRequestContext requestContext) {
+    if (requestContext.getRequest().getParamLength() == 2) {
+      final SendRawTransactionConditionalParameter conditions =
+          conditionalParameter(requestContext);
+      if (!withinRange(
+          blockchainQueries.get().headBlockNumber(),
+          conditions.getBlockNumberMin(),
+          conditions.getBlockNumberMax())) {
+        return getJsonRpcErrorResponse(
+            requestContext,
+            USER_SPECIFIED_CONDITIONS_NOT_MET,
+            "block number not within specified range");
```

Comment:
is it costly to add details about the block numbers here?

this apply also to the other error descriptions below

Reply:
added